logmiddleware.go

```diff
@@ -66,17 +63,8 @@ func removePort(host string) string {
 	return strings.Split(host, ":")[0]
 }
 
-func getBodyLength(myw readableResponseWriter) int {
-	if content := myw.Header().Get("Content-Length"); content != "" {
-		if length, err := strconv.Atoi(content); err == nil {
-			return length
-		}
-	}
-	return myw.Length()
-}
-
-func getReqID(logger *logrus.Logger, headers http.Header) string {
-	if requestID := headers.Get("X-Request-Id"); requestID != "" {
+func getReqID(logger *logrus.Logger, getHeader func(string) string) string {
```

Comment:
why a func as an argument? getReqId could be the interface method instead of defining the GetHeader method.

anyway I'd bind the getReqID function with the interface passing it as an argument rather than a func as argument here.

Reply:
I forgot about this. It should use the `loggingContext` interface.